pkg/envoy_access_log_server/util.go

```diff
@@ -0,0 +1,22 @@
+package envoy_access_log_server
+
+import (
+	"fmt"
+
+	"github.com/golang/protobuf/ptypes"
+	pbduration "github.com/golang/protobuf/ptypes/duration"
+	"github.com/pkg/errors"
+)
+
+// Returns deterministic string representation of 'pbduration' - ns
+// panics in case of error
+func pbDurationDeterministicString(pbduration *pbduration.Duration) string {
+	if pbduration == nil {
+		return ""
+	}
+	duration, err := ptypes.Duration(pbduration)
+	if err != nil {
+		panic(errors.Wrap(err, "Unable to convert protobuf Duration to stdlib Duration"))
```

Comment:
I'd definitely avoid using `panic` or do we have a reason for it?
And if so, should we recover?

Reply:
refactored within https://github.com/seznam/slo-exporter/pull/37/commits/6575eeaed6b25c491f81c99cb7693d49f90fcced